features/authentication-framework/org.wso2.carbon.identity.application.authentication.framework.server.feature/resources/application-authentication.xml

```diff
@@ -168,12 +168,15 @@
             <Parameter name="windowSize">3</Parameter>
             <Parameter name="authenticationMandatory">true</Parameter>
             <Parameter name="enrolUserInAuthenticationFlow">true</Parameter>
-            <Parameter name="usecase">local</Parameter>
+            <Parameter name="usecase">userAttribute</Parameter>
+            <Parameter name="userAttribute">http://wso2.org/foursquare/claims/email</Parameter>
```

Comment:
Can you check the usage of `usecase` and `userAttribute` parameters? I can't understand the meaning of having `http://wso2.org/foursquare/claims/email` in here.

Reply:
If the 1st step is authenticated using local authenticator then these configurations won't get take part. 

If the 1st step is authenticated using federated authenticator then we could specify the usecase as associate or user attribute. 

If you choose use case as associate then the username will get from the associated local user's username.

If you choose usecase as userattribute then you could specify the userattribute claim as another parameter so that the username will be read from the claim you specified as userattribute.